controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -566,11 +570,26 @@ func (r *KubeadmControlPlaneReconciler) reconcileDelete(ctx context.Context, con
 	machinesToDelete := controlPlane.Machines.Filter(collections.Not(collections.HasDeletionTimestamp))
 	var errs []error
 	for i := range machinesToDelete {
-		m := machinesToDelete[i]
-		logger := log.WithValues("Machine", klog.KObj(m))
-		if err := r.Client.Delete(ctx, machinesToDelete[i]); err != nil && !apierrors.IsNotFound(err) {
-			logger.Error(err, "Failed to cleanup owned machine")
-			errs = append(errs, err)
+		machineToDelete := machinesToDelete[i]
+		log := log.WithValues("Machine", klog.KObj(machineToDelete))
+
+		// During KCP deletion we don't care about forwarding etcd leadership or removing etcd members.
+		// So we are removing the pre-terminate hook.
+		// This is important because after we went through the Machine deletion
+		// the etcd quorum will be broken and we can't forward etcd leadership or remove etcd members anymore.
```

Comment:
Nit: Etcd quorum breaks because we don't remove the members. The real reason is that we plan to delete the whole etcd cluster / all members.

```suggestion
		// This is important because when deleting KCP we will delete all members of etcd and its not possible
    // to forward etcd leadership without any member left.after we went through the Machine deletion
```